src/js/DatePicker.js

```diff
@@ -0,0 +1,436 @@
+                $target.trigger('click');
+                this.closeCalendar();
+            }
+        } else if (keyCode === keyCodes.PAGEUP) {
+            if ($target.is('a.ui-state-default')) {
+                keyEvent.preventDefault();
+                this.navigationManager.navigateOneMonth($target, 'prev', this.$datePickerContainer, this.$nextLink, this.$prevLink);
+            }
+        } else if (keyCode === keyCodes.PAGEDOWN) {
+            if ($target.is('a.ui-state-default')) {
+                keyEvent.preventDefault();
+                this.navigationManager.navigateOneMonth($target, 'next', this.$datePickerContainer, this.$nextLink, this.$prevLink);
+            }
+        } else if (keyCode === keyCodes.HOME) {
+            if ($target.is('a.ui-state-default')) {
+                keyEvent.preventDefault();
+
+                const $firstOfMonth = $target.closest('tbody').find('.ui-state-default').first();
+                if ($firstOfMonth) {
+                    $firstOfMonth.trigger('focus');
+                    this.highlightUtils.setHighlightState($firstOfMonth, $container);
+                }
+            }
+        } else if (keyCode === keyCodes.END) {
+            if ($target.is('a.ui-state-default')) {
+                keyEvent.preventDefault();
+
+                const $lastDay = $target.closest('tbody').find('.ui-state-default').last();
+                if ($lastDay) {
+                    $lastDay.trigger('focus');
+                    this.highlightUtils.setHighlightState($lastDay, $container);
+                }
+            }
+        }
+    }
+
+    /**
+     * Handle calendar close
+     */
+    closeCalendar () {
+        // Remove bound handler
+        this.$html.find('#ui-datepicker-div').off('keydown');
+        // Close the date picker
+        this.$dateInput.datepicker('hide');
+        // Return focus to the date input
+        this.$dateInput.trigger('focus');
+    }
+
+    /**
+     * Handle next month link click event
+     * @param {Event} event
+     */
+    handleNextLinkClicks (event) {
+        if (event !== undefined) {
+            event.preventDefault();
+        }
+
+        // Update the header elements after change of month
+        this.updateHeaderElements();
+        // Highlight date in new month
+        this.highlightUtils.prepHighlightState(this.$datePickerContainer);
+        // Refocus the next button (otherwise focus returns to date input)
+        this.$html.find('.ui-datepicker-next').trigger('focus');
+        // Re-hide the today button
+        this.$html.find('.ui-datepicker-current').addClass('u-display-none');
+    }
+
+    /**
+     * Handle previous month link click event
+     * @param {Event} event
+     */
+    handlePrevLinkClicks (event) {
+        if (event !== undefined) {
+            event.preventDefault();
+        }
+
+        // Update the header elements after change of month
+        this.updateHeaderElements();
```

Comment:
Feel like a lot of these comments are useless (not to mention repetitive which defeats the point of a method - see above) and end up hiding key information which might be in other comments. I've been skipping the comments when reading the PR because there's so many of them - but I shouldn't be. 

Here, we know it's going to "Update the header elements after change of month" because the method is called `updateHeaderElements`!

Reply:
Admittedly I do like to over-comment, purely so things are more obvious when revisited. I've removed some, and re-worded a couple.